server/game/cards/04.1-AtSK/RubyOfRhllor.js

```diff
@@ -15,7 +16,17 @@ class RubyOfRhllor extends DrawCard {
             handler: context => {
                 this.game.promptForCardName({
                     player: context.player,
-                    onSelect: (player, cardName) => this.selectCardName(player, cardName),
+                    onSelect: (player, cardName) => {
+                        this.game.resolveGameAction(
+                            GameActions.revealCards(context => ({
+                                cards: [...context.game.currentChallenge.loser.hand],
+                                player: context.player,
+                                revealWithMessage: false
+                            }))
+                            , context);
+                        this.selectCardName(player, cardName);
+                        return true;
+                    },
```

Comment:
Good fix - although I can see a small scenario where the card(s) would be discarded (within `selectCardName`) prior to the revealing their hand, and it causes a problem. If the named card is Alla Tyrell, her Interrupt should allow her to enter play prior to the reveal or discard's actually taking affect, which would prevent her from being discarded via this ability (as she's moved from the hand to in-play).

If you wanted to implement for that, it would likely need to be a combination of a `GameActions.Simultaneously` triggering both `GameActions.revealCards` & `GameActions.IfCondition`, with the if condition checking if any _revealed_ cards in that hand match, and if so, triggers a `GameActions.DiscardCard` on those cards. Happy to discuss privately more about how these GameActions work if you'd like!

Otherwise simply commenting a TODO here is also fine to just recognise that this will miss a specific scenario, and it can maybe be cleaned up later. :)